src/Cache/DataCache.php

```diff
@@ -0,0 +1,114 @@
+ *
+ * Copyright (c) 2004-2022, Ryan Parman, Sam Sneddon, Ryan McCue, and contributors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * 	* Redistributions of source code must retain the above copyright notice, this list of
+ * 	  conditions and the following disclaimer.
+ *
+ * 	* Redistributions in binary form must reproduce the above copyright notice, this list
+ * 	  of conditions and the following disclaimer in the documentation and/or other materials
+ * 	  provided with the distribution.
+ *
+ * 	* Neither the name of the SimplePie Team nor the names of its contributors may be used
+ * 	  to endorse or promote products derived from this software without specific prior
+ * 	  written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS
+ * AND CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @package SimplePie
+ * @copyright 2004-2022 Ryan Parman, Sam Sneddon, Ryan McCue
+ * @author Ryan Parman
+ * @author Sam Sneddon
+ * @author Ryan McCue
+ * @link http://simplepie.org/ SimplePie
+ * @license http://www.opensource.org/licenses/bsd-license.php BSD License
+ */
+
+namespace SimplePie\Cache;
+
+use InvalidArgumentException;
+
+/**
+ * Simplified PSR-16 Cache client for caching data arrays
+ *
+ * The methods names must be different, but should be compatible to the
+ * methods of \Psr\SimpleCache\CacheInterface.
+ *
+ * @package SimplePie
+ * @subpackage Caching
+ */
+interface DataCache
+{
+    /**
+     * Fetches a value from the cache.
+     *
+     * Equivalent to \Psr\SimpleCache\CacheInterface::get()
+     * <code>
+     * public function get(string $key, mixed $default = null): mixed;
+     * </code>
+     *
+     * @param string   $key     The unique key of this item in the cache.
+     * @param mixed    $default Default value to return if the key does not exist.
+     *
+     * @return array|mixed The value of the item from the cache, or $default in case of cache miss.
+     *
+     * @throws InvalidArgumentException
+     *   MUST be thrown if the $key string is not a legal value.
+     */
+    public function getData($key, $default = null);
```

Comment:
I think I should change the methods to camel_case.

Reply:
On another hand, switching to `camelCase` might give us opportunity to clean up the API. The new methods can use native types (e.g. `DateTimeImmutable` instead of `get_date`’s `int|string`) and once we drop PHP 5 support, native return typehints.

Reply:
Good point. Using the switch from snake_case to camelCase to introduce return types is a very good idea. But because this will require the merge of #743, the decision to switch to camelCase in v2 and complete new interfaces I would recommend to use snake_case for now. I put this idea to the roadmap #731.

Reply:
For `final` classes (or, I guess, `internal` ones too), we should not need to worry about [variance](https://www.php.net/manual/en/language.oop5.variance.php) so it should not matter if we add typehints later. That should allows to use camel case here now.

Reply:
Yes, you are right. But the decision to switch to camelCase has to be discussed and made first in another issue. Then I can still change the PR.